imap_processing/lo/l0/science_direct_events.py

```diff
@@ -76,45 +65,54 @@ class ScienceDirectEvents(LoBase):
 
     SHCOARSE: int
     COUNT: int
-    DATA: bitstring.Bits
+    DATA: str
     CHKSUM: int
-    TOF0: float
-    TOF1: float
-    TOF2: float
-    TOF3: float
-    TIME: float
-    ENERGY: float
-    POS: float
+    TOF0: np.ndarray
+    TOF1: np.ndarray
+    TOF2: np.ndarray
+    TOF3: np.ndarray
+    TIME: np.ndarray
+    ENERGY: np.ndarray
+    POS: np.ndarray
 
     def __init__(self, packet, software_version: str, packet_file_name: str):
         """Intialization method for Science Direct Events Data class."""
         super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
         self.parse_data(packet)
+        self.TOF0 = np.array([])
```

Comment:
This extends the individual arrays, there is an alternative of making a list of `ScienceDirectEvent` objects to keep the base structure of what you have and just putting multiples of those together instead if that is useful at all. `packet_events = [DirectEvent1, DirectEvent2, ...]`

Reply:
I'm not sure I'm following. The idea is to have a single packet worth of DEs stored in this object and then merged them all together when the CDF is created